service/timestreamwrite/api_client.go

```diff
@@ -87,6 +97,16 @@ type Options struct {
 	// The HTTP client to invoke API calls with. Defaults to client's default HTTP
 	// implementation if nil.
 	HTTPClient HTTPClient
+
+	// The cache used to store discoevred endpoint. Defaults to 10.
+	CacheSize int64
```

Comment:
Nit this should be `EndpointDiscoveryCacheSize` to be more specific to its purpose.